frontend/src/components/users/useUsers.tsx

```diff
@@ -1,17 +1,46 @@
+import { Key, useEffect, useState } from "react";
 import User from "../user/User";
 
+export interface IUser {
+	username: string;
+	id: number;
+}
+
 const useUsers = () => {
-	const usersIds = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+	const [users, setUsers] = useState<[]>([]);
+
+	useEffect(() => {
+		fetch(`https://dummyjson.com/users/`)
+			.then(response => response.json())
+			.then(data =>
+				setUsers(
+					data.users.sort(function (a: any, b: any) {
```

Comment:
U have already function for filtering users, created by @masterHAWK99

Reply:
again, instructions unclear:
"Filter users alphabetically and display ordered list to the user."
you write about creating a filter and in the same line you say you want to display an *ordered* list of users. thus i assumed you wanted a SORTING algorithm.

Reply:
ye, but the sorting algorithm is implemented ealier by @masterHAWK99 and u can use it, and display, the same users the same algorithm

Reply:
i'm sorry, but that solution funnels the search query through search params. it's simpler to have a dedicated solution and avoid unnecessary clutter. i uploaded revised version with functional search.

Reply:
ok, but u change query params, when user type something, if not, that's not good solution, because if user refresh page or type manually something in url, the filter didn't work, show ep. default filtered users list, but in url is ep. "Stefan", and u don't looking for "Stefan".

Reply:
this solution does not deal with params at all. it's a separate solution. there is no need to make the browser generate a url with search params as far as searching userslist is concerned.